### ISSUE TITLE
Rule Change: Add `allowSelectors` option to "use-baseline" rule

### ISSUE THREAD

**Issue by Grapedge** [eslint/css#210]
### What rule do you want to change?

use-baseline

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```css
.foo {
  &:hover {
    color: red;
  }
}
```

### What does the rule currently do for this code?

> Selector 'nesting' is not a widely available baseline feature.

### What will the rule do after it's changed?

After the change, if a selector is included in `allowSelectors`, it will not report an error even if it's not a baseline feature.

```js
import css from '@eslint/css';

export default {
  files: ['**/*.css'],
  language: 'css/css',
  plugins: { css },
  languageOptions: {
    tolerant: true,
  },
  rules: {
    'css/use-baseline': [
      'error',
      { 
        available: 2021, 
        allowSelectors: ['nesting']  // ✅ New option
      }
    ],
  },
};
```

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

Our project uses PostCSS to process CSS nesting selector syntax, so the syntax in the Example code is valid for our environment.

We currently cannot set the 'css/use-baseline' rule to 'error' because it would generate numerous false positives in our codebase.

We hope to configure an allowlist of non-baseline selectors in ESLint to prevent errors when using PostCSS features.

I'm uncertain whether similar allow-list options should be supported for other CSS constructs like properties, at-rules, media conditions or types. Currently, I don't know if anyone actually needs these.

**Comment by nzakas** [comment]
Ah gotcha, I totally missed that part in the config. 

Can you help me understand a bit more: if you want to keep 2021 as your baseline, why do you want to specifically allow something that isn't supported? Wouldn't that defeat the purpose?

**Comment by rviscomi** [comment]
Stylelint actually did something similar: https://github.com/ryo-manba/stylelint-plugin-use-baseline/pull/22. I definitely get the usefulness of centrally allowlisting features, but I kind of wish developers didn't have to manage it in the first place.

@nzakas is there any precedent for ESLint adapting its behavior in the presence of other code transformation tools? For example, maybe it could automatically suppress warnings for things it knows will be handled by the other tool.

Also, I wonder if this is a common problem as there were a couple of negative reactions to https://github.com/eslint/css/issues/85, where I suggested adding support for this selector.

**Comment by EgoziE** [comment]
+1 for this, we have a strong need for this at Wix as well.

The ability to explicitly allow and deny certain CSS features is critical for us. We want to enforce a general baseline (widely available), but we also need an `ignore-list` for properties we don't want to be used and an `allow-list` for newer features we've approved for use.

Having both capabilities would give us the maximum flexibility to enforce our specific CSS standards at scale. 🕵🏼‍♂️

**Comment by nzakas** [comment]
For now I'd say let's not support regular expressions.

**Comment by ryo-manba** [comment]
For `allowProperties`, it might be better to support not just property names, but also property–value pairs.

In our plugin’s https://github.com/ryo-manba/stylelint-plugin-use-baseline/issues/37 need to handle values came up. 
While skipping entire properties might work, [@jeddy3’s comment](https://github.com/ryo-manba/stylelint-plugin-use-baseline/issues/37#issuecomment-3121532544) suggests that more flexible configurations are desirable.


For example:

- Allow only a specific value:

```js
allowProperties: { 'user-select': ['none'] }
  ```

- Allow any value (i.e. skip the entire property):
```js
allowProperties: { 'user-select': [/^.+$/]  }
```

More details can be found in [this comment](https://github.com/ryo-manba/stylelint-plugin-use-baseline/issues/37#issuecomment-3126620437).

**Comment by Grapedge** [comment]
I think it makes sense to support property–value pairs, and I’ll try implementing it in the following format. 
@nzakas, if we don’t need to implement this for now, or if you have any other suggestions, please let me know.

```ts
type UseBaselineOptions = [{
    available?: "widely" | "newly" | number;
    allowSelectors?: string[];
    allowAtRules?: string[];
    allowProperties?: Record<string, true | string[]>;
}];
```
Example:
```ts
allowProperties: {
  // Only allow specific values
  'user-select': ['none'],
  // Skip the entire property
  'backface-visibility': true,
}
```

**Comment by nzakas** [comment]
Let's keep `allowProperties` as just names for now.
